### ISSUE TITLE
PhpdocTypesOrder - Issue with nested generic types

### ISSUE THREAD

**Issue by julienfalque** [FriendsOfPHP/PHP-CS-Fixer#5461]
## Bug report

`phpdoc_types_order` rule does not handle nested generic types properly.

### Code snippet that reproduces the problem

```diff
 <?php
 
-/** @param SplObjectStorage<Foo<Bar>, Bar> $map */
+/** @param SplObjectStorage<, Bar|Foo<Bar>> $map */
 function foo($map) {}
```

**Comment by kubawerlos** [comment]
Let's handle https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4914 first.

**Comment by julienfalque** [comment]
This was fixed by #5630.
